src/lib/homestore_backend/heap_chunk_selector.cpp

```diff
@@ -96,6 +96,50 @@ csharedChunk HeapChunkSelector::select_chunk(homestore::blk_count_t count, const
     return vchunk.get_internal_chunk();
 }
 
+csharedChunk HeapChunkSelector::select_specific_chunk(const chunk_num_t chunkID) {
+    if (m_chunks.find(chunkID) == m_chunks.end()) {
+        // sanity check
+        LOGWARNMOD(homeobject, "No chunk found for ChunkID {}", chunkID);
+        return nullptr;
+    }
+
+    auto pdevID = VChunk(m_chunks[chunkID]).get_pdev_id();
```

Comment:
You've enforced `chunkID` be `const` to this stack, but not this? I would make `chunkID` non-const personally, but suggest consistency either way.

Reply:
sure, will use auto const pdevID.